modules/timeline.xql

```diff
@@ -61,11 +71,17 @@ let $output :=  if(exists($entry/atom:category)) then
             <a href="{$path}/{$entry/wiki:id/string()}" class="exist-read-more">Read more</a>
             <span class="exist-date">{
                 if ($age < xdt:dayTimeDuration("PT1H")) then
-                    minutes-from-duration($age) || " minutes ago"
+                    let $minutes := if (minutes-from-duration($age) = 1) then ' minute' else ' minutes' return
+                    minutes-from-duration($age) || $minutes || " ago"
+                    
                 else if ($age < xdt:dayTimeDuration("P1D")) then
-                    hours-from-duration($age) || " hours ago"
+                    let $hours := if (hours-from-duration($age) = 1) then ' hour' else ' hours' return
```

Comment:
Please have the return statement in the following line. This will be hard to understand and easy to overlook in a month.